packages/mobile/src/fiatExchanges/ProviderOptionsScreen.tsx

```diff
@@ -91,6 +116,20 @@ function ProviderOptionsScreen({ route, navigation }: Props) {
     })
   }, [])
 
+  const fetchResponse = useAsync(fetchLocationFromIpAddress, [])
+
+  useEffect(() => {
+    const { result, status } = fetchResponse
+
+    if (result && status === 'success') {
+      const { alpha2, state } = result
+      setUserLocation({ country: alpha2, state })
+    } else if (status === 'error') {
+      const alpha2 = countryCallingCode ? getRegionCodeFromCountryCode(countryCallingCode) : null
+      setUserLocation({ country: alpha2, state: null })
+    }
+  }, [fetchResponse.status])
```

Comment:
nit: another way to do it which doesn't need `useEffect` and the additional `useState`

```suggestion
const asyncUserLocation = useAsync(async () => {
    try {
      const { alpha2, state } = await fetchLocationFromIpAddress()
      return { country: alpha2, state }
    } catch (error) {
      const alpha2 = countryCallingCode ? getRegionCodeFromCountryCode(countryCallingCode) : null
      return { country: alpha2, state: null }
    }
  }, [])
  
  const userLocation = asyncUserLocation.result
```

Reply:
Good suggestion, thanks! Still getting the hang of the best usage patterns for `useAsync`... agree this is cleaner.